### PR TITLE
Implemented Explicit Waits

### PR DESCRIPTION
**Pull request by LanaLynx1994**
Created test "Create New Playlist"
Refactored all tests using Explicit waits (except issue with Profile Tests)

### CODE REVIEW

File: src/test/java/Homework20aka19.java [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
-import java.time.Duration;
-
-public class Homework19 extends BaseTest{
+public class Homework20aka19 extends BaseTest{
     @Test
     public void deletePlaylist() throws InterruptedException {
```
Comment (Berichicko) [inline src/test/java/Homework20aka19.java:9]: Please, delete `throws InterruptedException`

**Comment by Berichicko** [comment]
image

HINT

```
import java.util.UUID;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.testng.Assert;
import org.testng.annotations.Test;

public class ProfileTests extends BaseTest {

  @Test
  public void changeProfileName() {
    //navigateToPage();
    provideEmail("sviatlana.rysiavets@testpro.io");
    providePassword("nTtAZKUq");
    clickSubmit();

    wait.until(ExpectedConditions.urlContains(".app/#!/home"));
    driver.get("https://qa.koel.app/#!/profile");
    driver.navigate().refresh();

    String uniqueName = generateUniqueName();
    changeName(uniqueName);
    saveChanges();
    wait.until(ExpectedConditions.urlContains(".app/#!/home"));

    String profileName = getProfileName();
    Assert.assertEquals(profileName, uniqueName);
  }

  private String getProfileName() {
    WebElement profileName = wait.until(
        ExpectedConditions.visibilityOfElementLocated(By.cssSelector("span.name")));
    return profileName.getText();
  }

  private String generateUniqueName() {
    return UUID.randomUUID().toString().replace("-", "");
  }

  private void changeName(String name) {
    provideCurrentPassword("nTtAZKUq");
    enterNewName(name);
    saveChanges();
  }

  private void saveChanges() {
    WebElement saveButton = wait.until(
        ExpectedConditions.elementToBeClickable(By.className("btn-submit")));
    saveButton.click();
  }

  private void enterNewName(String name) {
    WebElement newName = wait.until(
        ExpectedConditions.presenceOfElementLocated(By.id("inputProfileName")));
    newName.clear();
    newName.sendKeys(name);
  }

  private void provideCurrentPassword(String password) {
    WebElement currentPassword = wait.until(
        ExpectedConditions.visibilityOfElementLocated(By.id("inputProfileCurrentPassword")));
    currentPassword.clear();
    currentPassword.sendKeys(password);
  }
}
```

**Comment by LanaLynx1994** [comment]
Doesn't work. After renaming and saving Koel doesn't return to home page. So it fails here:
wait.until(ExpectedConditions.urlContains(".app/#!/home"));

**Comment by LanaLynx1994** [comment]
I changed it to wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show"))); and it worked!

**Pull request merged**